src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PoParser.cs

```diff
@@ -55,31 +61,32 @@ public IEnumerable<CultureDictionaryRecord> Parse(TextReader reader)
 
         private static string Unescape(string str)
         {
-            StringBuilder sb = null;
+            var builder = default(Utf16ValueStringBuilder);
```

Comment:
I would just check for `\` in the input string. (IndexOf is fast, it's vectorized). If not return the string. Otherwise create the ZString, check the other usages in Orchard, it needs to be disposed.

Reply:
I'm not sure why we need to check? BTW the disposing in my mind after the tests pass

Reply:
The idea is that you check that the `builder` is initialized in lots of places, to prevent initializing it if there is nothing to unescape.

My suggestion is to first check if there is anything to encode, by looking for `\` in the whole string first. If there is not you return, if there is then you create the builder and don't check anymore.

Looking for `\` is very fast as I explained.

Reply:
Example:

```c#
if (!str.Contains('\\'))
{
    return str;
}

using var builder = ZString.CreateStringBuilder();
... // no more checking, just use builder
```

Reply:
I already did that two weeks ago but the few tests were broken, I found the issue and fix it